### ISSUE TITLE
Ignore taps on showcase child

### ISSUE THREAD

**Issue by FabioSpano** [SimformSolutionsPvtLtd/flutter_showcaseview#340]
Hi,

I'm using the latest version on this git repo (commit 2402f14) and I'm noticing a strange behaviour. When I tap the widget that the Showcase is highlighting it triggers the Widget's onTap callback instead of proceeding to the next showcase.

For example I have this Card:
```
Showcase(
    key: _blablaKey,
    title: "Blabla",
    description: 'Blabla',
    child: Padding(
      padding: EdgeInsets.fromLTRB(15, 8, 15, 8),
      child: GestureDetector(
        onTap: () async {
            Navigator.of(context).push(CupertinoPageRoute(builder: (context) {
              return InitialChoice();
            }));
        },
        child: Card(
          elevation: 5,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(15.0),
          ),
          child: Center(
                Text("Armocromia",
                    style: GoogleFonts.quicksand(
                        color: Colors.black,
                        fontWeight: FontWeight.bold, fontSize: 20)),
            ),
        ),
      ),
    ),
  ),
```

Tapping the Card will trigger the onTap function of the GestureDetector widget.

How can I just ignore the onTap function if the showcase is active?

**Comment by FabioSpano** [comment]
Update: 

Using commit f5b348e fixed the issue, so it must be something related to the last commits.

**Comment by ujas-m-simformsolutions** [comment]
Hey @FabioSpano, I tested both commits and I got the same behavior, GestureDetector gets triggered in both commits. But, to resolve your issue you can use the `onStart` and `onComplete` function available in the `ShowCaseWidget`. They provide index and key of the showcase so you can just compare the index of the showcase when it is active and disable the GestureDetector.

**Comment by ujas-m-simformsolutions** [comment]
Closing this issue for now. If this doesn't resolve your issue then feel free to reopen this issue.

**Comment by ablbol** [comment]
> Hey @FabioSpano, I tested both commits and I got the same behavior, GestureDetector gets triggered in both commits. But, to resolve your issue you can use the `onStart` and `onComplete` function available in the `ShowCaseWidget`. They provide index and key of the showcase so you can just compare the index of the showcase when it is active and disable the GestureDetector.

Could you please show code on how to do that. Thanks

**Comment by Ujas-Majithiya** [comment]
You can check the main.dart in the example app, it contains usage of it or you can check [this](https://github.com/SimformSolutionsPvtLtd/flutter_showcaseview/issues/345#issuecomment-1443381592) comment it contains minimal code for this

**Comment by ablbol** [comment]
> You can check the main.dart in the example app, it contains usage of it or you can check [this](https://github.com/SimformSolutionsPvtLtd/flutter_showcaseview/issues/345#issuecomment-1443381592) comment it contains minimal code for this

I need the code that will disable taping on target while the showcasing is active.  This is the code that I saw in the example and it does not disable any tapping.

```
onStart: (index, key) {
            log('onStart: $index, $key');
          },
          onComplete: (index, key) {
            log('onComplete: $index, $key');
            if (index == 4) {
              SystemChrome.setSystemUIOverlayStyle(
                SystemUiOverlayStyle.light.copyWith(
                  statusBarIconBrightness: Brightness.dark,
                  statusBarColor: Colors.white,
                ),
              );
            }
          },
```

How do I disable tapping on targets while showcasing is running?  That is my question. Thanks

**Comment by Ujas-Majithiya** [comment]
I was just directing you to feature which you can use to implement it. As you are passing the gesture dector from your end, showcaseview doesn't come into the scope.

Anyways, you have to maintain a flag and update the flag to false once the index from onStart callback matches to your desired showcase. So when flag becomes false, pass null to gesture dector. And it will not trigger the tap.

**Comment by ablbol** [comment]
[Ujas-Majithiya](https://github.com/Ujas-Majithiya) how about if we make this a feature in the component.  You can create an overlay on top of the target to disable tapping on the target.  What do you think?

**Comment by ujas-m-simformsolutions** [comment]
@ablbol This feature is already available and parameter names are `onToolTipClick` and `onTargetClick`. You can check all list of parameters in the documentation.

**Comment by ablbol** [comment]
@Ujas-Majithiya I noticed that when I use onTargetClick then sometimes this function gets called and sometimes it does not get called.  Also when it is not called then the target onTap function gets called which I want to avoid.  My goal is that when the target gets called, then the onTap for the target gets ignored which is not happening most of the time.

**Comment by Abdelazeem777** [comment]
It is not working here as well!!

**Comment by ujas-m-simformsolutions** [comment]
@Abdelazeem777  Ignoring is fixed by #351 PR. We will try to release it as soon as possible.
